liquidprompt

```diff
@@ -2971,6 +2993,48 @@ _lp_git_staged_lines() {
     lp_vcs_staged_d_lines=$lp_git_diff_shortstat_d_lines
 }
 
+# Get the remote name for the current branch.
+# NOTE: Assumes that `_lp_find_vcs` and `_lp_vcs_branch` (costly functions) have been called.
+_lp_git_remote() {
+    lp_vcs_remote=""
+
+    # If there is no branch…
+    if [[ -z "$lp_vcs_branch" ]]; then
+        # … then there is no remote.
+        return 1
+    fi
+
+    # If we don't have a VCS root…
+    if [[ -z "$lp_vcs_root" ]]; then
+        # … we cannot find the config file.
+        return 1
+    fi
+
+    local file="${lp_vcs_root}/.git/config"
+    local delimiter=" = " # With spaces.
+    local in_branch=0
+    # We test for $line in the loop is here to ensure that we read the last line,
+    # even if the file does not ends with a \n.
+    # This bypass a known behavior of the C standard, not fixed in POSIX.
+    while IFS='' read -r line || [[ -n "$line" ]] ; do
+        # If we are in the branch config section.
+        if [[ "$line" == "[branch \"${lp_vcs_branch}\"]" ]]; then
+            in_branch=1
+        elif [[ "$line" == *"remote${delimiter}"* ]] ; then
+            if [[ "$in_branch" ]]; then
+                # Remove first part until delimiter.
+                lp_vcs_remote="${line#*"${key}${delimiter}"}"
+                return 0
+            else
+                in_branch=0
+            fi
+        fi
+    done <"$file"
```

Comment:
This block clearly does work. But do we need it to? I timed it:
```bash
$ time git config branch.master.remote
origin

real   	0m0.004s
user   	0m0.003s
sys    	0m0.001s

$ time _lp_git_remote

real   	0m0.002s
user   	0m0.001s
sys    	0m0.001s
```
The speed difference isn't that significant. And this isn't a feature enabled by default.

If you want to keep this design, I think we should have some tests to check this nested config object parsing.

```suggestion
    if lp_vcs_remote="$(git config "branch.${lp_vcs_branch}.remote")";
        return 0
    fi
```

Reply:
Considering it is still two times faster, I'm leaning toward keeping the more complex implementation. I gave a try to a test in a42f518, but _lp_find_vcs doesn't seem to work in that context (if you have any clue…).

Reply:
It never loads the directory because the list of enabled VCSs is never compiled. You need to run `lp_activate` to make that happen:
```diff
diff --git a/tests/test_git.sh b/tests/test_git.sh
index 337e59a..bea2155 100644
--- a/tests/test_git.sh
+++ b/tests/test_git.sh
@@ -18,7 +18,10 @@ function test_git {
     LP_ENABLE_VCS_ROOT=0
     LP_ENABLE_VCS_REMOTE=1

-    wd="${SHUNIT_TMPDIR}/test_remote/"
+    PS1=""
+    lp_activate
+
+    wd="${SHUNIT_TMPDIR}/test_remote"
     mkdir -p "$wd"
     cd "$wd"

@@ -46,7 +49,7 @@ function test_git {
     git remote add notsoremote git@nojhan.github:nojhan/liquidprompt.git

     _lp_git_remote
-    assetEquals "Remote is found." "notsoremote." "$lp_vcs_remote"
+    assertEquals "Remote is found." "notsoremote." "$lp_vcs_remote"
 }

 . ./shunit2
```